test/integration/uncaught.spec.js

```diff
@@ -16,7 +16,7 @@ describe('uncaught exceptions', function () {
       assert.equal(res.stats.failures, 1);
 
       assert.equal(res.failures[0].fullTitle,
-        'uncaught "before each" hook');
+        'uncaught "before each" hook for "test"');
```

Comment:
where is this coming from?

Reply:
It's just a bonus improvement, that came for free when improving the handling of uncaught exceptions, since it now invokes the current test callback.

It's better to have the error message contain which test that failed.